frontend/src/components/details/PieChartComponent.jsx

```diff
@@ -9,36 +12,24 @@ import theme from 'resources/theme';
  * @author Parham Gandomkar, Julian Oelhaf, Irem Toroslu
  */
 const PieChartComponent = () => {
-    //const series = [17, 13, 3, 2, 1, 42, 21];
-    const series = getMaterialComposition();
-    //series.sort();
-    //series.reverse();
-    // commented the two lines, because otherwise the numbers don't match the labels
+    const series = getMaterialCompositionData();
+    const labels = getMaterialCompositionLabels();
+
     const options = {
-        maintainAspectRatio: false,
-        responsive: true,
+        //TODO
+        maintainAspectRatio: true,
         legend: {
-            position: 'bottom',
-            // labels: {
-            // boxWidth: 10
-            // }
-        },     
+            position: 'top',
+            fontSize: 10
+        },
         chart: {
             type: 'donut'
         },
-        legend: {
-            fontSize: 10,
-
+        pie: {
+            expandOnClick: true
         },
-        labels: [
-            'Transformer oil',
-            'Steel',
-            'Pressboard',
-            'Stainless steel',
-            'Aluminium',
-            'Silicon steel',
-            'Copper'
-        ],
+        labels: labels,
+        // TODO: do the materials stay the same? otherwise doesn't make sense to use fixed color / hardcode them
```

Comment:
@ScoutAtlas They will change (I am 90% sure about that) Maybe we can have a hash table for the most common material color mappings and randomly assign colors for other materials.